flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/java/BatchTableEnvironment.scala

```diff
@@ -178,4 +178,32 @@ class BatchTableEnvironment(
 
     registerTableFunctionInternal[T](name, tf)
   }
+
+  /**
+    * Scans a table from registered temporary tables and registered catalogs.
+    *
+    * The table to scan must be registered in the TableEnvironment or
+    * must exist in registered catalog in the TableEnvironment.
+    *
+    * Example:
+    *
+    * to scan a registered temporary table
+    * {{{
+    *   val tab: Table = tableEnv.scan("tableName")
+    * }}}
+    *
+    * to scan a table from a registered catalog
+    * {{{
+    *   val tab: Table = tableEnv.scan("catalogName.dbName.tableName")
+    * }}}
+    *
+    * @param tablePath The path of the table to scan.
+    * @throws TableException if no table is found using the given table path.
+    * @return The resulting [[Table]].
+    */
+  @throws[TableException]
+  def scan(tablePath: String): Table = {
```

Comment:
hi, @twalthr,  there already existed scan method which is `def scan(tableName: String)`, I added a scan method which is `def scan(catalogName: String, dbName: String, tableName: String)` in first commit. Fabian suggest that we could extend the previous scan(String) to accept varargs parameters. And We would need to push the implementation to the Scala / Java versions of BatchTableEnvironment and StreamTableEnvironment because varargs are handled differently by Scala and Java. In this way, we could keep the API more concise because we only need a single scan() method. I think it's a good idea, so I updated the pr in the second commit. Maybe I'm confused something there.  Any advice? @fhueske, @twalthr .